File: back-end/src/main/java/com/example/backend/service/VehicleService.java
package com.example.backend.service;

import com.example.backend.entities.Vehicle;
import com.example.backend.service.base.BaseService;

import java.util.List;
import java.util.Optional;

public interface VehicleService extends BaseService<Vehicle> {
    Optional<Vehicle> findByLicensePlate(String licensePlate);
    List<Vehicle> findByAvailable(boolean available);
    Optional<Vehicle> findByIdWithDriver(Long vehicleId);
    List<Vehicle> findAvailableVehicles();
    List<Vehicle> findVehiclesBySchool(Long schoolId);
    List<Vehicle> findNearbyVehicles(Double latitude, Double longitude, Double radiusInMeters);
    List<Vehicle> findActivelyOperatingVehicles();
    long countCompletedTrips(Long vehicleId);
    void updateLocation(Long vehicleId, Double latitude, Double longitude);
    void markAsAvailable(Long vehicleId);
    void markAsUnavailable(Long vehicleId);
    void assignToSchool(Long vehicleId, Long schoolId);
    void removeFromSchool(Long vehicleId);
    void performMaintenance(Long vehicleId, String maintenanceType, String notes);
    void completeMaintenance(Long vehicleId, String notes);
    List<Vehicle> findVehiclesWithStats();
} 

File: front-end/src/services/VehicleService.ts
import type { Vehicle } from "../core/entities/vehicle.entity"
import type { CreateVehicleRequest, UpdateVehicleRequest } from "../core/dto/vehicle.dto"

export interface VehicleService {
  findAll(): Promise<Vehicle[]>
  findById(id: number): Promise<Vehicle>
  create(vehicleRequest: CreateVehicleRequest): Promise<Vehicle>
  update(id: number, vehicleRequest: UpdateVehicleRequest): Promise<Vehicle>
  deleteVehicle(id: number): Promise<void>
  findByLicensePlate(licensePlate: string): Promise<Vehicle>
  findAvailableVehicles(): Promise<Vehicle[]>
  findVehiclesBySchool(schoolId: number): Promise<Vehicle[]>
  findNearbyVehicles(latitude: number, longitude: number, radiusInMeters: number): Promise<Vehicle[]>
  findActivelyOperatingVehicles(): Promise<Vehicle[]>
  countCompletedTrips(id: number): Promise<number>
  updateLocation(id: number, latitude: number, longitude: number): Promise<void>
  markAsAvailable(id: number): Promise<void>
  markAsUnavailable(id: number): Promise<void>
  assignToSchool(id: number, schoolId: number): Promise<void>
  removeFromSchool(id: number): Promise<void>
  performMaintenance(id: number, maintenanceType: string, notes: string): Promise<void>
  completeMaintenance(id: number, notes: string): Promise<void>
  findVehiclesWithStats(): Promise<Vehicle[]>
}

